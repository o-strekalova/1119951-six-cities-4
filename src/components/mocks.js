export const offersAll = [{
  id: `1010`,
  city: {
    location: {
      lat: 52.3909553943508,
      long: 4.85309666406198,
      zoom: 13,
    },
    name: `Amsterdam`,
  },
  location: {
    lat: 52.3909553943508,
    long: 4.85309666406198,
    zoom: 13,
  },
  pictures: [`img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/room.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`],
  preview: `img/apartment-01.jpg`,
  title: `Wood and stone place`,
  type: `room`,
  price: 999,
  isPremium: true,
  isFavorite: false,
  rating: 4.4,
  description: ``,
  bedrooms: 2,
  guests: 4,
  features: [],
  owner: {
    avatar: `https://api.adorable.io/avatars/128`,
    name: ``,
    isSuper: true,
    id: 111,
  },
}, {
  id: `2020`,
  city: {
    location: {
      lat: 52.3909553943508,
      long: 4.85309666406198,
      zoom: 13,
    },
    name: `Paris`,
  },
  location: {
    lat: 52.3909553943508,
    long: 4.85309666406198,
    zoom: 13,
  },
  pictures: [`img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/room.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`],
  preview: `img/apartment-01.jpg`,
  title: `Wood and stone place`,
  type: `room`,
  price: 999,
  isFavorite: false,
  isPremium: true,
  rating: 4.4,
  description: ``,
  bedrooms: 2,
  guests: 4,
  features: [],
  owner: {
    avatar: `https://api.adorable.io/avatars/128`,
    name: ``,
    isSuper: true,
    id: 222,
  },
}];

export const offersRaw = [{
  "id": 1010,
  "city": {
    "location": {
      "latitude": 52.3909553943508,
      "longitude": 4.85309666406198,
      "zoom": 13,
    },
    "name": `Amsterdam`,
  },
  "location": {
    "latitude": 52.3909553943508,
    "longitude": 4.85309666406198,
    "zoom": 13,
  },
  "images": [`img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/room.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`],
  "preview_image": `img/apartment-01.jpg`,
  "title": `Wood and stone place`,
  "type": `room`,
  "price": 999,
  "is_premium": true,
  "is_favorite": false,
  "rating": 4.4,
  "description": ``,
  "bedrooms": 2,
  "max_adults": 4,
  "goods": [],
  "host": {
    "avatar_url": `https://api.adorable.io/avatars/128`,
    "name": ``,
    "is_pro": true,
    "id": 111,
  },
}, {
  "id": 2020,
  "city": {
    "location": {
      "latitude": 52.3909553943508,
      "longitude": 4.85309666406198,
      "zoom": 13,
    },
    "name": `Paris`,
  },
  "location": {
    "latitude": 52.3909553943508,
    "longitude": 4.85309666406198,
    "zoom": 13,
  },
  "images": [`img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/room.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`],
  "preview_image": `img/apartment-01.jpg`,
  "title": `Wood and stone place`,
  "type": `room`,
  "price": 999,
  "is_favorite": false,
  "is_premium": true,
  "rating": 4.4,
  "description": ``,
  "bedrooms": 2,
  "max_adults": 4,
  "goods": [],
  "host": {
    "avatar_url": `https://api.adorable.io/avatars/128`,
    "name": ``,
    "is_pro": true,
    "id": 222,
  },
}];

export const reviews = [{
  id: 1110,
  avatar: `https://api.adorable.io/avatars/128`,
  name: `Angelina`,
  rating: 4.4,
  date: new Date(2020, 10, 5, 10),
  text: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.`,
}];

export const authInfo = {
  avatar: `https://api.adorable.io/avatars/128`,
  email: `mail@mail.ru`,
  id: 1,
  isSuper: true,
  name: `Angelina`,
};
