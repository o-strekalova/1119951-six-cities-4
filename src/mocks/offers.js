const OFFERS_TITLES = [`Beautiful & luxurious apartment at great location`, `Wood and stone place`, `Canal View Prinsengracht`, `Nice, cozy, warm big bed apartment`];
const OFFERS_TYPES = [`apartment`, `room`, `house`, `hotel`];
const AVATAR_URL = `https://api.adorable.io/avatars/128`;
const COORDINATES = [
  [52.3909553943508, 4.85309666406198],
  [52.369553943508, 4.85309666406198],
  [52.3909553943508, 4.929309666406198],
  [52.3809553943508, 4.939309666406198]
];
const OFFERS_COUNT = 4;

const getRandomIntegerNumber = (min, max) => {
  return min + Math.floor(Math.random() * (max - min));
};

const getRandomArrayItem = (array) => {
  const randomIndex = getRandomIntegerNumber(0, array.length);
  return array[randomIndex];
};

const generateOffers = () => {
  const offers = [];

  for (let i = 0; i < OFFERS_COUNT; i++) {
    offers.push({
      id: i,
      coords: COORDINATES[i],
      pictures: [`img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/room.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`],
      title: getRandomArrayItem(OFFERS_TITLES),
      type: getRandomArrayItem(OFFERS_TYPES),
      price: getRandomIntegerNumber(50, 1000),
      isPremium: Math.random() > 0.5,
      rating: getRandomIntegerNumber(0, 4) + Math.round(Math.random() * 10) / 10,
      description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.`,
      bedrooms: getRandomIntegerNumber(1, 10),
      guests: getRandomIntegerNumber(1, 20),
      features: [`Wi-Fi`, `Washing machine`, `Towels`, `Heating`, `Coffee machine`, `Baby seat`, `Kitchen`, `Dishwasher`, `Cabel TV`, `Fridge`],
      owner: {
        avatar: AVATAR_URL,
        name: `Angelina`,
        isSuper: Math.random() > 0.5,
      },
    });
  }

  return offers;
};

export const offers = generateOffers();
