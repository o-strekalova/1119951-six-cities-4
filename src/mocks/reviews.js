const AVATAR_URL = `https://api.adorable.io/avatars/128`;
const REVIEWS_COUNT = 11;

const getRandomIntegerNumber = (min, max) => {
  return min + Math.floor(Math.random() * (max - min));
};

const generateReviews = () => {
  const reviews = [];

  for (let i = 0; i < REVIEWS_COUNT; i++) {
    reviews.push({
      id: i,
      avatar: AVATAR_URL,
      name: `Angelina`,
      rating: getRandomIntegerNumber(0, 4) + Math.round(Math.random() * 10) / 10,
      date: new Date(),
      text: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.`,
    });
  }

  return reviews;
};

export const reviews = generateReviews();
