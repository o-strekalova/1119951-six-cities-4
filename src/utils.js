export const CardsListClass = {
  MAIN: `cities__places-list tabs__content`,
  PROPERTY: `near-places__list`,
};

export const CardClass = {
  MAIN: `cities__place-card`,
  PROPERTY: `near-places__card`,
};

export const SortType = {
  POPULAR: `Popular`,
  PRICE_FROM_LOW: `Price: low to high`,
  PRICE_FROM_HIGH: `Price: high to low`,
  TOP_RATED: `Top rated first`,
};

export const extend = (a, b) => {
  return Object.assign({}, a, b);
};

export const getRatingPercentage = (rating) => {
  return Math.round(rating) * 20;
};
