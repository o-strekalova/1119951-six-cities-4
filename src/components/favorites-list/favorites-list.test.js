import React from "react";
import renderer from "react-test-renderer";
import {Router} from "react-router-dom";
import FavoritesList from "./favorites-list.jsx";
import {offersAll, authInfo} from "../mocks";
import {AuthorizationStatus, noop} from "../../utils";
import history from "../../history";

it(`Render Favorites with offers`, () => {
  const tree = renderer
    .create(
        <Router
          history={history}
        >
          <FavoritesList
            authInfo={authInfo}
            authorizationStatus={AuthorizationStatus.AUTH}
            errorMessage={null}
            offers={offersAll}
            onCardTitleClick={noop}
            onFavoriteButtonClick={noop}
          />
        </Router>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it(`Render Favorites without offers`, () => {
  const tree = renderer
    .create(
        <Router
          history={history}
        >
          <FavoritesList
            authInfo={authInfo}
            authorizationStatus={AuthorizationStatus.AUTH}
            errorMessage={null}
            offers={[]}
            onCardTitleClick={noop}
            onFavoriteButtonClick={noop}
          />
        </Router>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it(`Render Favorites with ErrorMessage`, () => {
  const tree = renderer
    .create(
        <Router
          history={history}
        >
          <FavoritesList
            authInfo={authInfo}
            authorizationStatus={AuthorizationStatus.AUTH}
            errorMessage={`Action failed`}
            offers={[]}
            onCardTitleClick={noop}
            onFavoriteButtonClick={noop}
          />
        </Router>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
