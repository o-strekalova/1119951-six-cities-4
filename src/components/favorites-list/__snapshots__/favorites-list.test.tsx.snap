// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Favorites with ErrorMessage 1`] = `
<div
  className="page page--favorites-empty"
>
  <header
    className="header"
    id="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-linkheader__logo-link--active"
            href="/"
            onClick={[Function]}
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="../../img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <span
                className="header__nav-link header__nav-link--profile"
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <a
                  className="header__user-name user__name"
                  href="/favorites"
                  onClick={[Function]}
                >
                  mail@mail.ru
                </a>
              </span>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--favorites page__main--favorites-empty"
  >
    <div
      className="page__favorites-container container"
    >
      <section
        className="favorites favorites--empty"
      >
        <h1
          className="visually-hidden"
        >
          Favorites (empty)
        </h1>
        <div
          className="favorites__status-wrapper"
        >
          <b
            className="favorites__status"
          >
            Nothing yet saved.
          </b>
          <p
            className="favorites__status-description"
          >
            Save properties to narrow down search or plan yor future trips.
          </p>
        </div>
      </section>
    </div>
  </main>
  <footer
    className="footer container"
  >
    <a
      className="footer__logo-link"
      href="main.html"
    >
      <img
        alt="6 cities logo"
        className="footer__logo"
        height="33"
        src="img/logo.svg"
        width="64"
      />
    </a>
  </footer>
  <div
    style={
      Object {
        "backgroundColor": "red",
        "color": "white",
        "left": 0,
        "margin": "auto",
        "padding": "10px 20px",
        "position": "fixed",
        "right": 0,
        "textAlign": "center",
        "top": 0,
        "width": "200px",
      }
    }
  >
    Action failed
  </div>
</div>
`;

exports[`Render Favorites with offers 1`] = `
<div
  className="page"
>
  <header
    className="header"
    id="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-linkheader__logo-link--active"
            href="/"
            onClick={[Function]}
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="../../img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <span
                className="header__nav-link header__nav-link--profile"
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <a
                  className="header__user-name user__name"
                  href="/favorites"
                  onClick={[Function]}
                >
                  mail@mail.ru
                </a>
              </span>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--favorites"
  >
    <div
      className="page__favorites-container container"
    >
      <section
        className="favorites"
      >
        <h1
          className="favorites__title"
        >
          Saved listing
        </h1>
        <ul
          className="favorites__list"
        >
          <li
            className="favorites__locations-items"
          >
            <div
              className="favorites__locations locations locations--current"
            >
              <div
                className="locations__item"
              >
                <a
                  className="locations__item-link"
                  href="/"
                  onClick={[Function]}
                >
                  <span>
                    Amsterdam
                  </span>
                </a>
              </div>
            </div>
            <div
              className="favorites__places"
            >
              <article
                className="favorites__card place-card"
                onMouseOver={[Function]}
              >
                <div
                  className="place-card__mark"
                >
                  <span>
                    Premium
                  </span>
                </div>
                <div
                  className="favorites__image-wrapper place-card__image-wrapper"
                >
                  <a
                    href="#"
                  >
                    <img
                      alt="Wood and stone place"
                      className="place-card__image"
                      height="110"
                      src="img/apartment-01.jpg"
                      width="150"
                    />
                  </a>
                </div>
                <div
                  className="favorites__card-info place-card__info"
                >
                  <div
                    className="place-card__price-wrapper"
                  >
                    <div
                      className="place-card__price"
                    >
                      <b
                        className="place-card__price-value"
                      >
                        €
                        999
                      </b>
                      <span
                        className="place-card__price-text"
                      >
                        / night
                      </span>
                    </div>
                    <button
                      className="place-card__bookmark-button button"
                      onClick={[Function]}
                      type="button"
                    >
                      <svg
                        className="place-card__bookmark-icon"
                        height="19"
                        width="18"
                      >
                        <use
                          xlinkHref="#icon-bookmark"
                        />
                      </svg>
                      <span
                        className="visually-hidden"
                      >
                        To bookmarks
                      </span>
                    </button>
                  </div>
                  <div
                    className="place-card__rating rating"
                  >
                    <div
                      className="place-card__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "80%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <h2
                    className="place-card__name"
                  >
                    <a
                      href="/offer/1010/#header"
                      onClick={[Function]}
                    >
                      Wood and stone place
                    </a>
                  </h2>
                  <p
                    className="place-card__type"
                  >
                    Room
                  </p>
                </div>
              </article>
            </div>
          </li>
          <li
            className="favorites__locations-items"
          >
            <div
              className="favorites__locations locations locations--current"
            >
              <div
                className="locations__item"
              >
                <a
                  className="locations__item-link"
                  href="/"
                  onClick={[Function]}
                >
                  <span>
                    Paris
                  </span>
                </a>
              </div>
            </div>
            <div
              className="favorites__places"
            >
              <article
                className="favorites__card place-card"
                onMouseOver={[Function]}
              >
                <div
                  className="place-card__mark"
                >
                  <span>
                    Premium
                  </span>
                </div>
                <div
                  className="favorites__image-wrapper place-card__image-wrapper"
                >
                  <a
                    href="#"
                  >
                    <img
                      alt="Wood and stone place"
                      className="place-card__image"
                      height="110"
                      src="img/apartment-01.jpg"
                      width="150"
                    />
                  </a>
                </div>
                <div
                  className="favorites__card-info place-card__info"
                >
                  <div
                    className="place-card__price-wrapper"
                  >
                    <div
                      className="place-card__price"
                    >
                      <b
                        className="place-card__price-value"
                      >
                        €
                        999
                      </b>
                      <span
                        className="place-card__price-text"
                      >
                        / night
                      </span>
                    </div>
                    <button
                      className="place-card__bookmark-button button"
                      onClick={[Function]}
                      type="button"
                    >
                      <svg
                        className="place-card__bookmark-icon"
                        height="19"
                        width="18"
                      >
                        <use
                          xlinkHref="#icon-bookmark"
                        />
                      </svg>
                      <span
                        className="visually-hidden"
                      >
                        To bookmarks
                      </span>
                    </button>
                  </div>
                  <div
                    className="place-card__rating rating"
                  >
                    <div
                      className="place-card__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "80%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <h2
                    className="place-card__name"
                  >
                    <a
                      href="/offer/2020/#header"
                      onClick={[Function]}
                    >
                      Wood and stone place
                    </a>
                  </h2>
                  <p
                    className="place-card__type"
                  >
                    Room
                  </p>
                </div>
              </article>
            </div>
          </li>
        </ul>
      </section>
    </div>
  </main>
  <footer
    className="footer container"
  >
    <a
      className="footer__logo-link"
      href="main.html"
    >
      <img
        alt="6 cities logo"
        className="footer__logo"
        height="33"
        src="img/logo.svg"
        width="64"
      />
    </a>
  </footer>
</div>
`;

exports[`Render Favorites without offers 1`] = `
<div
  className="page page--favorites-empty"
>
  <header
    className="header"
    id="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-linkheader__logo-link--active"
            href="/"
            onClick={[Function]}
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="../../img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <span
                className="header__nav-link header__nav-link--profile"
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <a
                  className="header__user-name user__name"
                  href="/favorites"
                  onClick={[Function]}
                >
                  mail@mail.ru
                </a>
              </span>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--favorites page__main--favorites-empty"
  >
    <div
      className="page__favorites-container container"
    >
      <section
        className="favorites favorites--empty"
      >
        <h1
          className="visually-hidden"
        >
          Favorites (empty)
        </h1>
        <div
          className="favorites__status-wrapper"
        >
          <b
            className="favorites__status"
          >
            Nothing yet saved.
          </b>
          <p
            className="favorites__status-description"
          >
            Save properties to narrow down search or plan yor future trips.
          </p>
        </div>
      </section>
    </div>
  </main>
  <footer
    className="footer container"
  >
    <a
      className="footer__logo-link"
      href="main.html"
    >
      <img
        alt="6 cities logo"
        className="footer__logo"
        height="33"
        src="img/logo.svg"
        width="64"
      />
    </a>
  </footer>
</div>
`;
